paymentsheet/src/main/java/com/stripe/android/paymentelement/embedded/SharedPaymentElementViewModel.kt

```diff
@@ -100,39 +101,55 @@ internal class SharedPaymentElementViewModel @Inject constructor(
         intentConfiguration: PaymentSheet.IntentConfiguration,
         configuration: EmbeddedPaymentElement.Configuration,
     ): ConfigureResult {
-        return configurationHandler.configure(
-            intentConfiguration = intentConfiguration,
-            configuration = configuration,
-        ).fold(
-            onSuccess = { state ->
-                val previousConfiguration = confirmationStateHolder.state?.configuration
-                confirmationStateHolder.state = EmbeddedConfirmationStateHolder.State(
-                    paymentMethodMetadata = state.paymentMethodMetadata,
-                    selection = state.paymentSelection,
-                    initializationMode = PaymentElementLoader.InitializationMode.DeferredIntent(intentConfiguration),
-                    configuration = configuration,
-                )
-                customerStateHolder.setCustomerState(state.customer)
-                selectionHolder.set(
-                    selectionChooser.choose(
-                        paymentMethodMetadata = state.paymentMethodMetadata,
-                        paymentMethods = state.customer?.paymentMethods,
-                        previousSelection = selectionHolder.selection.value,
-                        newSelection = state.paymentSelection,
-                        previousConfiguration = previousConfiguration,
-                        newConfiguration = configuration,
+        return viewModelScope.async {
```

Comment:
Wrapping in the viewModelScope.async is the only change here. Everything is is just nesting + moving the logic that was nested into `handleLoadedState`.